### DISCUSSION TITLE
Measure depth at given pixel

### DISCUSSION THREAD

**Discussion by alex-klimov** [NVlabs/instant-ngp#624] [Q&A] [answered]
What is the recommended way to measure the depth at given pixel?

There is method determine_autofocus_target_from_pixel, which measure median depth around given pixel:
https://github.com/NVlabs/instant-ngp/blob/675bf4fdd2d8249bacafd1cadb7de25cd5aacd97/src/testbed.cu#L2320

This is not precise enough for my task - I need as exact depth at given pixel as trained network can give me.

**Comment by Tom94** [comment, marked as answer]
On latest master, `Testbed::determine_autofocus_target_from_pixel` is actually pixel-accurate, so you should be able to use it as-is.

**Reply by alex-klimov** [reply]
Tom, this is brilliant! It works perfectly.

The only question I have is to myself, where did last couple weeks of debugging went and why I didn't ask earlier...
